src/NServiceBus.Hosting.Windows/EndpointConfigurationExtensions.cs

```diff
@@ -0,0 +1,20 @@
+﻿namespace NServiceBus
+{
+    using Configuration.AdvanceExtensibility;
+
+    /// <summary>
+    /// Extentions for <see cref="EndpointConfiguration"/>.
+    /// </summary>
+    public static class EndpointConfigurationExtensions
+    {
+        /// <summary>
+        /// Defines the endpoint name in code thus overriding the previously assigned endpoint name.
+        /// </summary>
+        /// <param name="configuration">The configuration.</param>
+        /// <param name="endpointName">The endpoint name to be used.</param>
+        public static void DefineEndpointName(this EndpointConfiguration configuration, string endpointName)
+        {
+            configuration.GetSettings().Set("NServiceBus.Routing.EndpointName", endpointName);
```

Comment:
tbh I don't want people starting to use this key and fiddling in the settings, but from a versioning perspective it would probably make sense to have it to avoid changing it accidentally in a minor.